tools/percy/snapshots.js

```diff
@@ -60,18 +64,39 @@ const pagesToTest = [
   {
     url: '/live/',
     title: 'Live page',
+    navPage: true,
   },
 ];
 
+async function waitForNavUpdate(page, browser) {
+  try {
+    page.navPage &&
+      (await browser.waitFor(
+        (url) => {
+          const activeLink = document.querySelector('a[active=""]');
+
+          return activeLink && activeLink.getAttribute('href') === url;
+        },
+        {polling: 15, timeout: 5000},
+        page.url,
+      ));
+  } catch (err) {
+    console.log(`Couldn't find active link for ${page.url}`);
+  }
+}
+
 // A script to navigate our app and take snapshots with Percy.
 PercyScript.run(
   async (browser, percySnapshot) => {
+    // set the viewport to a desktop size
+    await browser.setViewport({height: 1024, width: 1280});
```

Comment:
@Robdel12 we're also using a .percy.conf.yml which sets dimensions
https://github.com/GoogleChrome/web.dev/blob/master/tools/percy/.percy.conf.yml

Should we remove this line so the yml file can set mobile and tablet sizes?

Reply:
@robdodson That's a good question! There's a bit of a disconnect between those two settings. And I admit its a little confusing.

The snapshot options are passed along to our browsers in our infrastructure. The widths being passed there are to tell our browsers to resize to that and then capture the screenshot.

This line in the test script is changing the width of the _test_ browser, which can impact how the DOM renders (at this moment Percy is capturing a DOM snapshot & not a screenshot). If the applications DOM changes between screen sizes, you'll have to make some changes to the script for accurate screenshots. At the moment Percy makes the assumption that the DOM snapshot is the same across widths and CSS media queries is what changes the pages look.

Reply:
ah I see